### PR TITLE
Feature/explain

### PR DESCRIPTION
**Pull request by craigerm**
Hey there,

I've put together an explain feature for genghis (with tests). This is for issue #60.

Right now the the feature is available as a checkbox in the advanced search area. If you have a better idea for how to enable this in the UI let me know. I think in the advanced area is a good idea because it's out of the way.

I've included the php change too but it's been quite a while since I did any php work so it was pretty hacky.

Any feedback would be great!

### CODE REVIEW

**Comment by bobthecow** [comment]
Thanks for working on this!

I wonder if it would work better as a button than a checkbox, since "explain" feels more like a different thing you could do with the same form...

screen shot 2013-07-04 at 12 02 21 am

**Comment by craigerm** [comment]
No worries! It was something I was needing lately.

Good idea, that does look better.  I wasn't sold on the checkbox - and changing it will clean up some of the view logic.

I'll make the change and let you know,

File: src/php/Genghis/Models/Collection.php [hunk 1]
```diff
@@ -145,6 +145,14 @@ public function findDocuments($query = null, $page = 1)
             ->limit(Genghis_Api::PAGE_LIMIT)
             ->skip($offset);
 
+        if ($explain == true) {
+          # Treat explain result as an iterator with a fake ID on the first
+          # item.
```
Comment (bobthecow) [inline src/php/Genghis/Models/Collection.php]: PHP supports comments like this, but does not recommend them. `//` is a better option.

Comment (craigerm) [inline src/php/Genghis/Models/Collection.php]: Thanks for the heads up. I must have been in ruby mode - I've changed it to match the other comments.

**Comment by craigerm** [comment]
I've changed it to be a button, which feels much more natural - and saves a click which is nice.
It also cleans up some of the backbone views which is a nice benefit.

**Comment by bobthecow** [comment]
Hey, thanks for the updates. The button definitely works better.

Sorry about the delay. I decided it would be great to take it a step further and make it a first-class feature and API endpoint. I've merged this branch into `develop`, and added shuffled things around a bit to make room for it: bfc91e97cc723f1c9580710a68e44b4dd8d204da

Explain will be in the next release! Thanks again :)

**Pull request closed without merging**